src/mappings/entities/mapConfidentialAsset.ts

```diff
@@ -0,0 +1,134 @@
+import { Codec } from '@polkadot/types/types';
+import { ConfidentialAsset, ConfidentialVenue, EventIdEnum, ModuleIdEnum } from '../../types';
+import { getBigIntValue, getBooleanValue, getTextValue, getNumberValue } from '../util';
+import { Attributes, HandlerArgs } from './common';
+
+export const getAuditorsAndMediators = (
+  item: Codec
+): Pick<Attributes<ConfidentialAsset>, 'auditors' | 'mediators'> => {
+  const { auditors, mediators } = JSON.parse(item.toString());
+
+  return {
+    auditors,
+    mediators,
+  };
+};
+
+const handleConfidentialAssetCreated = async (blockId: string, params: Codec[]): Promise<void> => {
+  const creatorId = getTextValue(params[0]);
+  const assetId = getTextValue(params[1]);
+  const { auditors, mediators } = getAuditorsAndMediators(params[2]);
+
+  await ConfidentialAsset.create({
+    id: assetId,
+    assetId,
+    creatorId,
+    auditors,
+    mediators,
+    totalSupply: BigInt(0),
+    venueFiltering: false,
+    createdBlockId: blockId,
+    updatedBlockId: blockId,
+  }).save();
+};
+
+const handleConfidentialAssetIssued = async (blockId: string, params: Codec[]): Promise<void> => {
+  const assetId = getTextValue(params[1]);
+  const totalSupply = getBigIntValue(params[3]);
+
+  const asset = await ConfidentialAsset.get(assetId);
+
+  asset.totalSupply = totalSupply;
+  asset.updatedBlockId = blockId;
+
+  await asset.save();
+};
+
+const handleVenueCreated = async (blockId: string, params: Codec[]): Promise<void> => {
+  const creatorId = getTextValue(params[0]);
+  const venueId = getNumberValue(params[1]);
+
+  await ConfidentialVenue.create({
+    id: `${venueId}`,
+    venueId,
+    creatorId,
+    createdBlockId: blockId,
+    updatedBlockId: blockId,
+  }).save();
+};
+
+const handleVenuesAllowed = async (blockId: string, params: Codec[]): Promise<void> => {
+  const assetId = getTextValue(params[1]);
+  const venuesAllowed = JSON.parse(params[2].toString()).map(val => getNumberValue(val));
+
+  const asset = await ConfidentialAsset.get(assetId);
+
+  if (asset.allowedVenuesId) {
+    asset.allowedVenuesId = [...asset.allowedVenuesId, ...venuesAllowed];
```

Comment:
Does this need deduplication? I think the chain allows set to be triggered even though no venues change.

Doesn't look as nice, but might be safer:
```ts
...new Set([...allowedVenues, ...venuesAllowed])
```